src/broker/Manager.cc

```diff
@@ -839,16 +857,16 @@ RecordVal* Manager::MakeEvent(ValPList* args, zeek::detail::Frame* frame) {
 
             auto num_args = func->GetType()->Params()->NumFields();
 
-            if ( num_args != args->length() - 1 ) {
-                Error("bad # of arguments: got %d, expect %d", args->length(), num_args + 1);
+            if ( num_args != argc - 1 ) {
+                Error("bad # of arguments: got %d, expect %d", argc - 1, num_args + 1);
                 return rval;
             }
 
             rval->Assign(0, func->Name());
             continue;
         }
 
-        const auto& got_type = (*args)[i]->GetType();
+        const auto& got_type = (*it)->GetType();
```

Comment:
We already have an alias for that:

```suggestion
        const auto& got_type = arg_val->GetType();
```